src/screens/CreatePost/index.js

```diff
@@ -506,14 +509,23 @@ const CreatePost = () => {
         });
       }
 
-      await createPost(data);
+      const post = await createPost(data);
       if (params.onRefresh && typeof params.onRefresh === 'function') {
         params.onRefresh();
       }
-      showMessage({
-        message: StringConstant.createPostDone,
-        type: 'success'
-      });
+      if (post.code === 200) {
+        showMessage({
+          message: StringConstant.createPostDone,
+          type: 'success'
+        });
+        setLoadingPost(false);
+      } else {
+        showMessage({
+          message: StringConstant.createPostFailedGeneralError,
+          type: 'danger'
+        });
+        setLoadingPost(false);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

You've correctly handled the success and error cases of the `createPost` function by setting `loadingPost` to false in both scenarios. This ensures that the loading state is properly reset after the post operation completes, regardless of whether it was successful or not.
```diff
+ const post = await createPost(data);
+ if (params.onRefresh && typeof params.onRefresh === 'function') {
+   params.onRefresh();
+ }
+ if (post.code === 200) {
+   showMessage({
+     message: StringConstant.createPostDone,
+     type: 'success'
+   });
+   setLoadingPost(false);
+ } else {
+   showMessage({
+     message: StringConstant.createPostFailedGeneralError,
+     type: 'danger'
+   });
+   setLoadingPost(false);
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->